parsers/GE.py

```diff
@@ -3,167 +3,246 @@
+            zone_key,
         )
 
-        # Collect the data into a list of dictionaries, then validate and
-        # return it.
-        production_mixes = (
-            {
-                "datetime": timestamp.to_pydatetime().replace(tzinfo=TIMEZONE),
-                "production": {
-                    "gas": production_mix["gas"],
-                    "hydro": production_mix["hydro"],
-                    "wind": production_mix["wind"],
-                    "solar": production_mix["solar"],
-                },
-                "source": URL.netloc,
-                "zoneKey": zone_key,
-            }
-            for timestamp, production_mix in table.items()
-        )
-        return [
-            validation.validate(
-                production_mix,
-                logger,
-                remove_negative=True,
-                floor=MINIMUM_PRODUCTION_THRESHOLD,
+    tables = pandas.read_excel(
+        response.content,
+        header=None,
+        # labels column + the 24 columns of hourly data
+        usecols=[1] + list(range(3, 27)),
+        # drop every 11th row (separation between tables)
+        skiprows=lambda x: x % 11 == 0,
+    )
+
+    production_breakdown_list = ProductionBreakdownList(logger)
+    for _i, daily_table in reversed(
+        tuple(tables.groupby(np.arange(len(tables)) // 10))
+    ):
+        day = daily_table.iloc[1][1]
+        timestamps = pandas.date_range(start=day, freq="1H", periods=24)
+
+        # zoom in on fields of interest
+        itemised_production_data = daily_table.iloc[4:8].set_index(1)
+        itemised_production_data.columns = timestamps
+
+        for timestamp, production in itemised_production_data.dropna(
+            axis="columns", how="all"
+        ).items():
+            dt = timestamp.to_pydatetime().replace(tzinfo=TIMEZONE)
+
+            production_mix = ProductionMix()
+            production_mix.add_value("gas", production["Thermal Power Plants"])
+            production_mix.add_value("hydro", production["Hydro Power Plants"])
+            production_mix.add_value("solar", production["Wind Power Plants"])
+            production_mix.add_value("wind", production["Solar Power Plants"])
+
+            production_breakdown_list.append(
+                zoneKey=zone_key,
+                datetime=dt.astimezone(timezone.utc),
+                source=SOURCE,
+                production=production_mix,
             )
-            for production_mix in production_mixes
-        ]
+    return _floor(
+        production_breakdown_list, floor=MINIMUM_PRODUCTION_THRESHOLD
+    ).to_list()
 
 
 def fetch_exchange(
-    zone_key1: str = "GE",
-    zone_key2: str = "TR",
+    zone_key1: ZoneKey,
+    zone_key2: ZoneKey,
     session: Session | None = None,
     target_datetime: datetime | None = None,
-    logger=getLogger(__name__),
-) -> list[dict] | dict:
+    logger: Logger = getLogger(__name__),
+) -> list[dict]:
```

Comment:
Just as a FYI, we will probably switch to using ENTSO-E directly instead of this source as it don't have any fixed timestamps to read and the fact that it's using ENTSO-E for historical data anyway which is polluting the database with both sources.